include/seqan3/core/detail/reflection.hpp

```diff
@@ -112,20 +112,18 @@ struct get_display_name
         for (; *name_ptr == ' '; ++name_ptr)
         {}
 
-        for (unsigned i = 0; i < tmp.size(); ++i, ++name_ptr)
-        {
-            tmp[i] = *name_ptr;
-        }
+        tmp.insert(tmp.begin(), name_ptr, name_ptr + tmp.capacity());
```

Comment:
not .size()?

Reply:
my default constructed `small_string` has size `0` so this would not work then. (Renes default constructed constexpr_string had size = capacity.). The `get_display_name_size_v<type>` with whom I templetize `small_string` sets the capacity exaclty to the length of `name_ptr`.

Reply:
> my default constructed small_string has size 0 so this would not work then.

Why? If it's size 0, you don't need to copy...

> The get_display_name_size_v<type> with whom I templetize small_string sets the capacity exaclty to the length of the string pointed to by name_ptr

Yes, but if there is nothing to copy there is nothing to copy and default initialisation is ok.

Reply:
I refactored the code further. It should be clear now what happens